sec_certs_page/dashboard/callbacks/dashboard.py

```diff
@@ -0,0 +1,332 @@
+        ),
+        inputs=dict(
+            save_clicks=Input(f"{prefix}-save-dashboard-btn", "n_clicks"),
+        ),
+        state=dict(
+            dashboard_id=State(f"{prefix}-current-dashboard-id", "data"),
+            dashboard_name=State(f"{prefix}-dashboard-name-input", "value"),
+            chart_configs=State(f"{prefix}-chart-configs-store", "data"),
+        ),
+        prevent_initial_call=True,
+    )
+    def save_dashboard(save_clicks, dashboard_id, dashboard_name, chart_configs):
+        print(f"[SAVE] Callback triggered - save_clicks: {save_clicks}")
+        print(f"[SAVE] chart_configs keys: {list((chart_configs or {}).keys())}")
+
+        if not save_clicks:
+            return dict(
+                selector_options=no_update,
+                selector_value=no_update,
+                current_dashboard_id=no_update,
+            )
+
+        user_id = get_current_user_id()
+        if not user_id:
+            return dict(
+                selector_options=no_update,
+                selector_value=no_update,
+                current_dashboard_id=no_update,
+            )
+
+        # Use chart_configs as the sole source of truth
+        chart_ids = list((chart_configs or {}).keys())
+        print(f"[SAVE] Using chart_ids: {chart_ids}")
+
+        charts = []
+        for i, chart_id in enumerate(chart_ids):
+            # First try to get from chart_configs
+            config = (chart_configs or {}).get(chart_id)
+            if config:
+                try:
+                    chart = Chart.from_dict(config)
+                    chart.order = i
+                    charts.append(chart)
+                    print(f"[SAVE] Chart {chart_id}: loaded from chart_configs")
+                except (KeyError, ValueError) as e:
+                    print(f"[SAVE] Chart {chart_id}: failed to parse from configs - {e}")
+                    continue
+            else:
+                # Try to get from predefined chart registry
+                predefined = chart_registry.get_predefined(chart_id)
+                if predefined and predefined.config:
+                    chart = predefined.config
+                    chart.order = i
+                    charts.append(chart)
+                    print(f"[SAVE] Chart {chart_id}: loaded from predefined registry")
+                else:
+                    print(f"[SAVE] Chart {chart_id}: not found in configs or registry")
+
+        print(f"[SAVE] Total charts to save: {len(charts)}")
+
+        # Load existing dashboard or create new one
+        if dashboard_id:
+            dashboard = dashboard_manager.get_dashboard(dashboard_id)
+            if dashboard and dashboard.user_id == user_id:
+                dashboard.name = dashboard_name or "Untitled Dashboard"
+                dashboard.charts = charts
+            else:
+                # Dashboard not found or doesn't belong to user, create new
```

Comment:
Do we really want to create one if it does not exist? How could it arise that dashboard_id is present yet it does not exist on our end?

I would just error out in this case. If not, the code below is also wrong:
```python
is_new_dashboard = dashboard_id is None
```
while in this branch a new dashboard is created yet it is the `dashboard_id != None` branch.